FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -66,7 +67,63 @@ public class AzureBlobJobStore : IJobStore
             _renewLockTimer.Elapsed += async (sender, e) => await RenewJobLockLeaseAsync();
         }
 
-        public async Task<bool> AcquireJobLock(CancellationToken cancellationToken = default)
+        public async Task<Job> AcquireJobAsync(CancellationToken cancellationToken = default)
+        {
+            var lockAcquired = await TryAcquireJobLockAsync(cancellationToken);
+            if (!lockAcquired)
+            {
+                _logger.LogWarning("Start job conflicted. Failed to acquire job lock.");
+                throw new StartJobFailedException("Start job conflicted. Will skip this trigger.");
+            }
+
+            var activeJobs = await GetActiveJobsAsync(cancellationToken);
+            Job job = activeJobs.FirstOrDefault();
+            if (job?.Status == JobStatus.Succeeded)
```

Comment:
If job get from active jobs is succeeded, can we check other active jobs rather than directly return null to create a new one?

Reply:
Good suggestion. Now we will complete all completed jobs until we find a new/failed job